### PR TITLE
Implement calculator

### PR DESCRIPTION
**Pull request by biadars**

### CODE REVIEW

File: CalculatorTests/Tests.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        public void AddManyNumbers()
+        {
+            int answer = CalculateAnswer("+", new[] {1, 2, 3, 4, 5, 6});
+            Assert.AreEqual(21, answer);
+        }
+
+        [Test]
+        public void ThrowException()
+        {
+            Assert.Throws<IndexOutOfRangeException>(() => CalculateAnswer("+", new int[] {}));
```
Comment (biadars) [inline CalculatorTests/Tests.cs:34]: I'm not sure this exception immediately conveys what's going wrong. Wrapping it in a custom exception explaining that the calculator does not accept empty lists might be more helpful for whoever triggers it.

File: Program.cs [hunk 2]
```diff
@@ -5,11 +5,91 @@ namespace Calculator
...
+    {
+      var op = AskForOperator();
+      var n = AskForNumberArray(op);
+      var a = CalculateAnswer(op, n);
+
+      Console.WriteLine("The answer is: " + a);
+      Console.WriteLine();
+    }
+
+    private static string AskForOperator()
```
Comment (biadars) [inline Program.cs:33]: This function does not just ask for the operator, it also reads it. I think a name like AskForAndReadOperator would make the code a bit clearer, especially when reading PerformOneCalculation

File: Program.cs [hunk 3]
```diff
@@ -5,11 +5,91 @@ namespace Calculator
...
+      for (int index = 0; index < count; index++)
+      {
+        numbers[index] = AddNumber($"Please enter number {index + 1}: ");
+      }
+      return numbers;
+    }
+
+    private static int AddNumber(string message)
+    {
+//      Console.Write(message);
```
Comment (biadars) [inline Program.cs:57]: I don't think you meant to keep this line

File: Program.cs [hunk 4]
```diff
@@ -5,11 +5,91 @@ namespace Calculator
...
+
+      int[] numbers = new int[count];
+      for (int index = 0; index < count; index++)
+      {
+        numbers[index] = AddNumber($"Please enter number {index + 1}: ");
+      }
+      return numbers;
+    }
+
+    private static int AddNumber(string message)
```
Comment (biadars) [inline Program.cs:55]: I think using add to mean append in this context might get confusing. This function also doesn't append the number, it just reads it, so I think ReadNumber would be more accurate

File: Program.cs [hunk 5]
```diff
@@ -5,11 +5,91 @@ namespace Calculator
...
+      int number;
+
+      int.TryParse(Console.ReadLine(), out number);
+
+      return number;
+    }
+
+    public static int CalculateAnswer(string op, int[] numbers)
+    {
+      // Returns 0 if there aren't any numbers.
```
Comment (biadars) [inline Program.cs:68]: I don't think this is true. If there are no numbers you'll get an index out of bound exception from the next line. You'll probably need to specifically check for that case.

File: Program.cs [hunk 6]
```diff
@@ -5,11 +5,91 @@ namespace Calculator
...
+      int answer = numbers[0];
+
+      int index = 0;
+      foreach (int number in numbers)
```
Comment (biadars) [inline Program.cs:72]: You can use numbers.Skip(1) here to avoid having to use the index conditional below

File: Program.cs [hunk 7]
```diff
@@ -5,11 +5,91 @@ namespace Calculator
...
+        }
+        else
+        {
+          if (op == "/")
+            answer = answer / number;
+          else if (op == "+")
+            answer = answer + number;
+          else if (op == "-")
+            answer = answer - number;
+          else
```
Comment (biadars) [inline Program.cs:86]: You seem to have missed out implementing *

File: Program.cs (hunk 6 above)
Comment (biadars) [reply Program.cs:72]: You also won't need the index variable anymore if you do this

File: CalculatorTests/Tests.cs [hunk 8]
```diff
@@ -0,0 +1,37 @@
+﻿using System;
```
Comment (biadars) [inline CalculatorTests/Tests.cs:1]: These are good test cases, it's useful to cover edge cases for the list size. I think it's worth adding some cases for the other operands as well though, and trying to find edge cases for the numbers provided. For example, what happens if the user enters * and 100 0.5? What about / and 100 0?
